sound/soc/sof/pm.c

```diff
@@ -497,7 +499,7 @@ int snd_sof_suspend(struct device *dev)
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 	int ret;
 
-	if (snd_sof_dsp_d0i3_on_suspend(sdev)) {
+	if (sdev->d0i3_on_suspend) {
```

Comment:
also based on @keyonjie 's findings, don't we need consistency in the HDA-specific resume parts as well?

Reply:
@plbossart Yes, I will rebase after his PR is merged.

Reply:
that seems just wrong. We are going to have one patch from Keyon 'fixing' stuff, and one patch from Ranjani updating the fix for the same lines.